### PR TITLE
Move enum helpers

### PR DESCRIPTION
**Pull request by cmcantalupo**
Requires merging #1163 first.

### CODE REVIEW

File: src/record.cpp [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+        };
+        auto it = result_map.find(hint);
+        if (it == result_map.end()) {
+            throw Exception("Profile::hint_hame(): Invalid hint",
+                            GEOPM_ERROR_INVALID, __FILE__, __LINE__);
+        }
+        return it->second;
+    }
+
+    uint64_t hint_type(std::string hint_name)
```
Comment (bgeltz) [inline src/record.cpp]: const std::string&

File: src/ProfileTracer.cpp [hunk 2]
```diff
@@ -98,7 +98,7 @@ namespace geopm
             // This is a call to format the event column
             // Store the event type for the next call
             event_type = value;
-            result = ApplicationSampler::event_name((int)value);
+            result = event_name((int)value);
```
Comment (cmcantalupo) [inline src/ProfileTracer.cpp:101]: doesn't need a cast any more.

File: src/record.hpp [hunk 3]
```diff
@@ -85,6 +86,23 @@ namespace geopm
                                  ///         parameter).
     };
 
+    /// @brief Format an event_e type as a string.
+    std::string event_name(int event_type);
```
Comment (bgeltz) [inline src/record.hpp:90]: If these methods are going to be called a million times a second for populating the profile trace, should you be using string references wherever possible?

**Review by bgeltz — CHANGES_REQUESTED** [review]

File: src/record.hpp (hunk 3 above)
Comment (cmcantalupo) [reply src/record.hpp:90]: Not for return values, that is resolved by return value optimization:

https://en.cppreference.com/w/cpp/language/copy_elision

File: src/record.cpp (hunk 1 above)
Comment (cmcantalupo) [reply src/record.cpp]: thanks!

**Review by bgeltz — DISMISSED** [review]

**Review by bgeltz — APPROVED** [review]

**Pull request merged**